src/server/server.ts

```diff
@@ -97,14 +91,15 @@ app.get('/*', async (req, res) => {
 
     const articleId = req.params[0] || defaultId;
 
-    const template = await readTemplate();
     const key = await getConfigValue<string>("capi.key");
     const imageSalt = await getConfigValue<string>('apis.img.salt');
     const resp = await fetch(capiEndpoint(articleId, key), {});
     const capi = await resp.text();
+    const parsedCapi = parseCapi(capi);
+    const articleContainer = ArticleContainer(includesTweets(parsedCapi))
+    const template = renderToString(articleContainer)
```

Comment:
So if I understand this correctly the template is rendered to a string, then we inject more components, should the article container be allowed to have children instead?

Reply:
I can refactor to pass in child components, that why we would only use `renderToString` once and not twice.

Reply:
Although I'll probably wait until this is merged in before refactoring https://github.com/guardian/apps-rendering/pull/101 @JamieB-gu